scipy/stats/mstats_basic.py

```diff
@@ -382,9 +382,10 @@ def pearsonr(x,y):
     df = n-2
     if df < 0:
         return (masked, masked)
-
-    (mx, my) = (x.mean(), y.mean())
-    (xm, ym) = (x-mx, y-my)
+        
+    # Mask arrays to make same size
+    (xm, ym) = (ma.MaskedArray(x, m), ma.MaskedArray(y, m))
```

Comment:
Not sure what you mean it's not used - it will change the calculation of r_den on line 391 (not quite sure how to add that to this comment).